src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Serialization/TagHelperDescriptorJsonConverter.cs

```diff
@@ -25,7 +25,15 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
             }
 
             // Required tokens (order matters)
-            var descriptorKind = reader.ReadNextStringProperty(nameof(TagHelperDescriptor.Kind));
+            var (hash, descriptorKind) = reader.ReadPropertyAndHash(nameof(TagHelperDescriptor.Kind));
+
+            if (hash != null &&
+                TagHelperDescriptorCache.TryGetDescriptor(hash.Value, out var descriptor))
```

Comment:
Definitely  would  have expected to  see this cache impl  in this PR

Reply:
Just created a separate one with cache + refactor of the memory cache for easier readability: https://github.com/dotnet/aspnetcore-tooling/pull/2308